pkg/store/proxy_merge.go

```diff
@@ -796,6 +804,8 @@ func sortWithoutLabels(set []*storepb.SeriesResponse, labelsToRemove map[string]
 }
 
 func (l *eagerRespSet) Close() {
+	l.wg.Wait()
+
 	if l.closeSeries != nil {
 		l.closeSeries()
```

Comment:
Should this be above the `l.wg.Wait()` so that in case of an error the goroutine would end its work early? Because now even if there is some error then we will wait until this is done finishing downloading everything just to discard all data.